rust/src/storage/file/mod.rs

```diff
@@ -134,6 +134,17 @@ impl StorageBackend for FileStorageBackend {
     async fn delete_obj(&self, path: &str) -> Result<(), StorageError> {
         fs::remove_file(path).await.map_err(StorageError::from)
     }
+
+    async fn delete_objs(&self, paths: &[String]) -> Result<(), StorageError> {
```

Comment:
I think `paths: &[&str]` would be a slight better signature.

Reply:
I originally used `&[&str]` but later met some problem when calling this method from `vacuum`... It said that: temporary value created here. So I think maybe we should use `String` instead of `&str` here?

Reply:
Ideally, I think the method should be defined as a generic function taking `impl Iterator<AsRef<str>>` as input. But this trait is being used as trait object, so we can't use generic here, which makes things really annoying. I also think `&[&str]` would have been a cleaner interface, but it will force us to allocate both `Vec<String>` and `Vec<&str>` in vacuum method, which has performance overheads :( I also don't really have a good suggestion on how to workaround this issue here.

Reply:
@Dandandan Do you have any suggestion on this? Or we could go ahead with `&[String]` for now?

Reply:
I think it's fine 👍